src/bundle/Resources/views/themes/admin/ui/component/dropdown/dropdown.html.twig

```diff
@@ -0,0 +1,117 @@
+
+{% set choices_flat = choices
+    | map(choice => choice.choices is defined ? choice.choices : [choice])
+    | reduce((carry, choices) => carry|merge(choices), [])
+%}
+{% set preferred_choices = preferred_choices|default([]) %}
+{% set min_search_items = min_search_items|default(5) %}
+{% set custom_form = custom_form is defined ? custom_form : true %}
+{% set translation_domain = translation_domain|default(false) %}
+{% set value = value is defined ? value : null %}
+{% set multiple = multiple is defined ? multiple : false %}
+{% set no_items = choices_flat|length == 0 %}
+
+{% macro get_translated_label(label, translation_domain) %}
+    {{ translation_domain is same as(false) ? label : label|trans({}, translation_domain) }}
+{% endmacro %}
+
+{% macro get_translated_placeholder(placeholder, translation_domain) %}
+    {% set default_placeholder = 'dropdown.placeholder'|trans|desc("Choose an option") %}
+    {% set placeholder = placeholder|default(default_placeholder) %}
+    {% set label = placeholder %}
+
+    {{ _self.get_translated_label(placeholder, translation_domain )}}
```

Comment:
Aren't you translating the placeholder twice? It looks like:
1. you assign translated label to `default_placeholder`
2. you pass it as `placeholder` to `get_translated_label` where it is translated again when `translation_domain` is not false.

I guess that from the backend you are getting a placeholder that is not translated?
I'm also not sure whether we need `get_translated_label` as a separate macro for such a small thing.

Reply:
On one hand it's small thing, but it's used in five different places, its shorthand if, etc., I think it looks neater with macro